### ISSUE TITLE
Can I dynamically insert Descriptions into fields?

### ISSUE THREAD

**Issue by bentham7246** [otykier/TabularEditor#738]
**What are you trying to achieve?**

I have a bunch of columns in my model that I would like to add descriptions to.   I know I can manually enter these into the descriptions property box for each column and I have played around with using the advanced scripting to automate this.

On the scripting side I'm trying to find a way to dynamically add descriptions to my columns based on a lookup to another table in my model.  

For example

This is table1 from my model.   I need to add descriptions against these columns

Units Sold | Gross Margin | Category 1 | Category 2
-- | -- | -- | --
100 | 50 | AAA | XXX
150 | 75 | BBB | YYY

![image](https://user-images.githubusercontent.com/33332692/105493243-09e68080-5cb1-11eb-866d-606161815e68.png)

I have another table (table2)  that contains 2 columns.  The first column contains the names of the fields from table1 and the second column contains the description of each field.  The table also has a measure called "columndescription" as follows: 
   ColumnDescription = SELECTEDVALUE(Table2[Description])

Column | Description
-- | --
Units Sold | The  number of units sold
Gross   Margin | Sales less Cost of Sales
Category 1 | Product   Category 1
Category   2 | Product Category 2

If possible I need a way to loop through the values in table2 and match it back to the field names in table1 so that I can bring back the corresponding description and add it to the property description.

I would just like to know if this is possible really before going any further.    I know I can export the properties, do a lookup in excel and import them back in but looking for something more automated.

**What have you tried so far?**

I've just been exploring the evaluatedax function and with that I'm able to spin up an output as follows:

var dax = "SUMMARIZECOLUMNS('Table2'[Column], " + string.Join(",", Selected.Measures.Select(m => "\"" + m.Name + "\", " + m.DaxObjectFullName).ToArray()) + ")";
EvaluateDax(dax).Output();

![image](https://user-images.githubusercontent.com/33332692/105493860-d9ebad00-5cb1-11eb-8164-2b595a32b1f9.png)

So that gives me a glimmer of hope.   I just need a way to be able to access the content of this output and link it back up to the columns properties in table1

Any help would be greatly appreciated

**Comment by otykier** [comment]
Instead of using a measure with SUMMARIZECOLUMNS, why not simply query the full 'Table2' table?

Use the `ExecuteReader` method to produce an [AmoDateReader](https://docs.microsoft.com/en-us/dotnet/api/microsoft.analysisservices.amodatareader?view=analysisservices-dotnet) that you can iterate through.

```csharp
var dax = "EVALUATE 'Table2'";
// ExecuteDax(dax).Output(); // Assuming this produces the same result as in your script... if not, just revert the definition of the dax variable above
var reader = ExecuteReader(dax);
while(reader.Read()) {
    var columnName = reader.GetString(0);
    var columnDescription = reader.GetString(1);
    var column = Model.Tables["Table1"].Columns.FindByName(columnName);
    if(column != null) column.Description = columnDescription;
}
```

See [Querying Analysis Services](https://docs.tabulareditor.com/Useful-script-snippets.html#querying-analysis-services) for more information.

**Comment by bentham7246** [comment]
Brilliant, that works a treat. I didn't realise I could use it like that.     Thanks!

**Comment by samfeinb** [comment]
C# is new to me... I found script to update all measure descriptions with dax expression which is super great for my business users to understand what is included in a calc (and also so they can learn some dax!) But when i tried to adapt to column descriptions, I came up with only errors.  I guess my wish list for the script to show in a column description is:

1.  source column name 
2. M Query table
3. M Query connection to source system
4. M Query source column?

Maybe this would go great in the "Useful Scripts?"  https://docs.tabulareditor.com/Useful-script-snippets.html   ?

Thank you for this amazing and indispensable tool.  I am looking forward to 3.0 and getting my company to pay!
